### PR TITLE
Custom header anchors

### PR DESCRIPTION
**Pull request by rafapaezbas**
This PR implements custom header ids in Marked using https://github.com/markedjs/marked-custom-heading-id, adds the header ids, corrects the indexes and implements the anchor navigation in pear-desktop-app.

### CODE REVIEW

**Review by subashsn — APPROVED** [review]

**Comment by subashsn** [comment]
Looks good, but suggest the code be reviewed by someone else on the team too as I don't have bg. on it yet.

File: lib/app-router.js [hunk 1]
```diff
@@ -16,9 +16,23 @@ customElements.define('app-router', class AppRouter extends HTMLElement {
         this.unload()
         document.documentElement.scrollTop = 0
         this.dataset.load = element.tagName.toLowerCase()
-        await element.load(page)
-        document.documentElement.scrollTop = 0
-        if (!opts.back) history.pushState({ pathname }, null, pathname)
+        await element.load(page, opts)
+
+        if (pathname.startsWith('/documentation') && !!opts.header) {
```
Comment (freeall) [inline lib/app-router.js]: Could you name this guard similar to `if (isDocumentationPage)`? A little unclear to me what this checks.

File: lib/app-router.js [hunk 2]
```diff
@@ -16,9 +16,23 @@ customElements.define('app-router', class AppRouter extends HTMLElement {
...
+          const anchor = opts.header
+          if (anchor) {
+            const element = this.routes['/documentation'].shadowRoot.getElementById(anchor)
+            element.scrollIntoView()
+            const elementY = Math.floor(element.getBoundingClientRect().y)
+            const headerHeight = 170
+            const extraScroll = 80
+            if (elementY < headerHeight + extraScroll) {
```
Comment (freeall) [inline lib/app-router.js]: Similar to this check, could you name this?

**Review by freeall — CHANGES_REQUESTED** [review]
A couple of things I wasn't sure of what does.

**Pull request merged**